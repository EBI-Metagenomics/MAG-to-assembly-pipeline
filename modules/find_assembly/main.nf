process FIND_PRIMARY_ASSEMBLY {
    label "python_based"
    
    input:
    path accessions

    output:
    path "*.links.tsv", emit: mag_assembly_pairs
    path "*.not_linked.tsv", emit: not_linked_mags

    script:
    """
    
    link_MAG_to_primary_metagenome_assembly.py \
        -i ${accessions} \
        -o ${accessions}.links.tsv \
        -p ${accessions}.putative.not_linked.tsv \
        -f ${accessions}.failed.not_linked.tsv \
        --download-folder fastas \
        --cleanup

    """
}