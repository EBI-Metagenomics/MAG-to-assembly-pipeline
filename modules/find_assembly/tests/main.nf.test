nextflow_process {

    name "Test Process FIND_PRIMARY_ASSEMBLY"
    script "../main.nf"
    process "FIND_PRIMARY_ASSEMBLY"

    test("Find assemblies for the test list of accessions") {

        when {
            params {
                portion_size = 2 
            }
            process {
                """
                input[0] = Channel.fromPath("${moduleDir}/tests/data/input_accessions.tsv", checkIfExists: true).splitText(by: params.portion_size, file: "portion")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
