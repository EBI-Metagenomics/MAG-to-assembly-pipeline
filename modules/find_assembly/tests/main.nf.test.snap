{
    "Find assemblies for the test list of accessions": {
        "content": [
            {
                "0": [
                    "portion.1.links.tsv:md5,7fe78c6e1fa3fb43751142bf93f92c3b",
                    "portion.2.links.tsv:md5,a36c984839f0d228135335852300e994"
                ],
                "1": [
                    [
                        "portion.1.failed.not_linked.tsv:md5,3dc8bdbaa4fb8705e606fafdca7f9392",
                        "portion.1.putative.not_linked.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ],
                    [
                        "portion.2.failed.not_linked.tsv:md5,7cf3bc7703fe76e435100dbac329fdf0",
                        "portion.2.putative.not_linked.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "mag_assembly_pairs": [
                    "portion.1.links.tsv:md5,7fe78c6e1fa3fb43751142bf93f92c3b",
                    "portion.2.links.tsv:md5,a36c984839f0d228135335852300e994"
                ],
                "not_linked_mags": [
                    [
                        "portion.1.failed.not_linked.tsv:md5,3dc8bdbaa4fb8705e606fafdca7f9392",
                        "portion.1.putative.not_linked.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ],
                    [
                        "portion.2.failed.not_linked.tsv:md5,7cf3bc7703fe76e435100dbac329fdf0",
                        "portion.2.putative.not_linked.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "23.04.3"
        },
        "timestamp": "2024-08-15T14:52:20.533170631"
    }
}